src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -540,7 +532,11 @@ public Person generatePerson(int index, long personSeed) {
 
         // TODO - export is DESTRUCTIVE when it filters out data
         // this means export must be the LAST THING done with the person
-        Exporter.export(person, finishTime, exporterRuntimeOptions);
+        if (Generator.entityManager == null || isAlive) {
+          // This if-statement prevents dead patients from being exported during
+          // fixed demographics runs.
+          Exporter.export(person, finishTime, exporterRuntimeOptions);
+        }
```

Comment:
Are you sure we need to do this? The check criteria should handle this, especially if you properly set some of the `public boolean` attributes on this class.